conans/client/build/meson.py

```diff
@@ -29,9 +30,18 @@ def __init__(self, conanfile, backend=None, build_type=None, append_vcvars=False
         self._append_vcvars = append_vcvars
 
         self._os = self._ss("os")
+
         self._compiler = self._ss("compiler")
+        if not self._compiler:
+            conan_v2_behavior("compiler setting should be defined.",
+                              v1_behavior=self._conanfile.output.warn)
+
         self._compiler_version = self._ss("compiler.version")
+
         self._build_type = self._ss("build_type")
+        if not self._build_type:
```

Comment:
If there are setters for ``build_type``, you don't want to raise here, but to delay the check to the ``configure()`` step